packages/next-admin/src/components/Filters.tsx

```diff
@@ -0,0 +1,70 @@
+"use client";
+import clsx from "clsx";
+import { useEffect, useState } from "react";
+import { useRouterInternal } from "../hooks/useRouterInternal";
+import { FilterWrapper, ModelName } from "../types";
+import Badge from "./Badge";
+
+type FiltersProps<T extends ModelName> = {
+  filters?: FilterWrapper<T>[];
+} & React.HTMLAttributes<HTMLDivElement>;
+
+const Filters = <T extends ModelName>({
+  filters,
+  ...props
+}: FiltersProps<T>) => {
+  const { router, query } = useRouterInternal();
+
+  const fetchUrlFilter = () => {
+    if (query?.filters) {
+      const urlFilter = JSON.parse(query.filters) as string[];
```

Comment:
Add a try / catch in case the filters query param is modified by the user to a non JSON string